File: Problem_03_largest_prime_factor/problem_3.py
import datetime
import math

# Simple function for primeness test
def is_prime(number):
    import math

    limit = int(math.sqrt(number)) + 1

    if number == 2:
        return True
    if number % 2 == 0:
        return False
    for i in range(3, limit, 2):
        if number % i == 0:
            return False

    return True


large_number = 600851475143
limit_large_number = int(math.sqrt(large_number))
answer = -1

time_start = datetime.datetime.now()

for i in range(2, limit_large_number):
    if large_number % i == 0 and is_prime(i):  # short circuit evaluation FTW!
        answer = i

time_end = datetime.datetime.now()

print("Result:", answer)
print("Microseconds elapsed:", time_end - time_start)

File: Problem_03_largest_prime_factor/problem_3.rs
use std::time::SystemTime;

fn is_prime(number: i64) -> bool {
    let limit = (number as f64).sqrt() as i64 + 1;
    let mut factor_exists = false;
    
    for i in 2..limit {
        if number % i == 0 {
            factor_exists = true;
            break;
        }
    }

    !factor_exists
}

fn main() {
    let large_number = 600851475143;
    let limit = (large_number as f64).sqrt() as i64 + 1;    

    let mut result = 0;

    let time_start = SystemTime::now();

    for i in (3..limit).rev() {
        if large_number % i == 0 && is_prime(i){
            result = i;
            break;
        }
    }

    match time_start.elapsed() {
        Ok(elapsed) => {
            println!("Calc duration: {}", elapsed.as_secs());
            println!("Subsec nanos: {}", elapsed.subsec_nanos());
        }
        Err(e) => {
            println!("Error: {:?}", e);
        }
    }

    println!("Result: {}", result);
}
